lib/components/form/call-taker/date-time-options.js

```diff
@@ -0,0 +1,211 @@
+ * - when a user changes the date or time, "leave now" (if selected) will
+ *   automatically switch to "depart at".
+
+ * @type {Object}
+ */
+export default class DateTimeOptions extends Component {
+  state = {}
+
+  componentDidMount () {
+    this._startAutoRefresh()
+  }
+
+  componentWillUnmount () {
+    this._stopAutoRefresh()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const {departArrive} = nextProps
+    // If departArrive has been changed to leave now, update time input to now.
+    if (departArrive === 'NOW' && !this.props.departArrive !== 'NOW') {
+      this._updateTimeInput()
+    }
+  }
+
+  _updateTimeInput = (time = moment()) =>
+    this.setState({timeInput: time.format(OTP_API_TIME_FORMAT)})
+
+  _startAutoRefresh = () => {
+    const timer = window.setInterval(this._refreshDateTime, 1000)
+    this.setState({ timer })
+  }
+
+  _stopAutoRefresh = () => {
+    window.clearInterval(this.state.timer)
+    this.setState({timer: null})
+  }
+
+  _refreshDateTime = () => {
+    const now = moment()
+    this._updateTimeInput(now)
+    const dateTimeParams = momentToQueryParams(now)
+    // Update query param if the current time has changed (i.e., on minute ticks).
+    if (dateTimeParams.time !== this.props.time) {
+      this.props.setQueryParam(dateTimeParams)
+    }
+  }
+
+  _setDepartArrive = evt => {
+    const {value: departArrive} = evt.target
+    if (departArrive === 'NOW') {
+      const now = moment()
+      // If setting to leave now, update date/time and start auto refresh to keep
+      // form input in sync.
+      this.props.setQueryParam({
+        departArrive,
+        ...momentToQueryParams(now)
+      })
+      if (!this.state.timer) {
+        this._startAutoRefresh()
+      }
+    } else {
+      // If set to depart at/arrive by, stop auto refresh.
+      this._stopAutoRefresh()
+      this.props.setQueryParam({ departArrive })
+    }
+  }
+
+  handleDateChange = evt => this.handleDateTimeChange({ date: evt.target.value })
+
+  /**
+   * Handler that should be used when date or time is manually updated. This
+   * will also update the departArrive value if need be.
+   */
+  handleDateTimeChange = params => {
+    const {departArrive: prevDepartArrive} = this.props
+    // If previously set to leave now, change to depart at when time changes.
+    if (prevDepartArrive === 'NOW') params.departArrive = 'DEPART'
+    this.props.setQueryParam(params)
+  }
+
+  /**
+   * Select input string when time input is focused by user (for quick changes).
+   */
+  handleTimeFocus = evt => evt.target.select()
+
+  handleTimeChange = evt => {
+    if (this.state.timer) this._stopAutoRefresh()
+    const timeInput = evt.target.value
+    const date = moment().startOf('day').format('YYYY-MM-DD')
+    const parsedTime = moment(date + 'T' + timeInput, SUPPORTED_TIME_FORMATS)
+    this.handleDateTimeChange({ time: parsedTime.format(OTP_API_TIME_FORMAT) })
+    this.setState({timeInput})
```

Comment:
I still think we should handle the input time in this PR so it is displayed in the right format. How about saving the input as timestamp, so that we render it with the configured format?

Reply:
I've commented some code in 4342a38fda36a4d1e6f2a77412b340c55e14cc10 to clarify the design intent. If there is any feedback from TriMet about usability we will address at that point, but for now I think we need to move forward on this PR.